platform/stm32f7xx/qspi.c

```diff
@@ -75,14 +81,36 @@ static uint32_t get_address_size(uint32_t address);
 static event_t cmd_event;
 static event_t rx_event;
 static event_t tx_event;
+static event_t st_event;
 
 status_t hal_error_to_status(HAL_StatusTypeDef hal_status);
 
+static status_t dma_disable(DMA_Stream_TypeDef *dma)
+{
+    // Unset the DMA Enable bit.
+    dma->CR &= ~DMA_SxCR_EN;
+
+    uint attempts = 0;
+
+    while (dma->CR & DMA_SxCR_EN) {
+
+        thread_sleep(1);
```

Comment:
Will add a comment.

Basically, unsetting the DMA Enable bit isn't good enough to disable the DMA controller since xfers might still be in progress. The solution is to unset the enable bit, then wait for it to actually become disabled.